nipype/interfaces/spm/base.py

```diff
@@ -126,28 +126,29 @@ def scans_for_fnames(fnames, keep4d=False, separate_sessions=False):
 
 class Info(PackageInfo):
     """Handles SPM version information
+
+    If you use `SPMCommand.set_mlab_paths` to set alternate entries for
+    matlab_cmd, paths, and use_mcr, then you will need to use the same entries
+    to any call in the Info class to maintain memoization. Otherwise, it will
+    default to the parameters in the `getinfo` function below.
```

Comment:
I'm conceptually a little displeased with this memoization principle, but in practice I imagine nobody's using multiple MATLABs and SPMs within a single nipype script, so it's probably fine.

Reply:
actually i think this exactly handles the multiple matlab/spm scenario. let's say i have two different versions of spm. i can explicitly use Info to retrieve them (yes, it will loose the first memory on collecting the second one).

the situation, i think, boils down to this.

`SPMCommand.set_mlab_paths` is a class method, so every instance derived will respect it. and i can adjust any given instance to change its command/path.

`Info` only has class methods and we would somehow need to detect a change. this method decouples the change detection saying the only memory it has is when you pass the same parameters.

an alternative was the previous augmented version to set the `Info` attributes everytime `set_mlab_paths` is called, retrieve that information and store it and always return that unless its changed again with get_info call. 

i don't feel strongly about either. just think that the latter seems more magical.

Reply:
Agreed, in the sense that this will be correct. The issue I was seeing with multiple versions I have is that you're running a MATLAB instance every time the cache is invalidated to get the information.

But it really doesn't matter. If people start complaining about slowdowns, we can consider keeping a memoizing dict keyed on a 3-tuple. This is probably fine, though.